pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -337,6 +359,67 @@ func (f *framework) RunScorePlugins(pc *PluginContext, pod *v1.Pod, nodes []*v1.
 	return pluginToNodeScoreMap, nil
 }
 
+// RunNormalizeScorePlugins runs the normalize score plugins. It should be called after
+// RunScorePlugins with the PluginToNodeScoreMap result. It then modifies the map with
+// normalized scores. It returns a non-success Status if any of the normalize score plugins
+// returns a non-success status.
+func (f *framework) RunNormalizeScorePlugins(pc *PluginContext, pod *v1.Pod, scores PluginToNodeScoreMap) *Status {
+	ctx, cancel := context.WithCancel(context.Background())
+	errCh := schedutil.NewErrorChannel()
+	workqueue.ParallelizeUntil(ctx, 16, len(f.scoreWithNormalizePlugins), func(index int) {
+		pl := f.scoreWithNormalizePlugins[index]
+		nodeScoreList, ok := scores[pl.Name()]
+		// During framework initialization, we make sure that NormalizeScore plugins are
```

Comment:
We do not need L375~378 since we make sure it exists. We could add the comment to L371. The code will be more clear.

Reply:
I suggested we add this check in an earlier round of comments, I agree that in practice this may not happen, however since this is a public function and this map is passed down to the function, in theory there could be a bug in the code path leading to here that results in the score list of a plugin not being added or somehow removed from the map. I would rather the system throw an error message than crash in this case.

One thing we can do to avoid those extra checks and simplify the code in generic_scheduler is to make both RunNormalizeScorePlugins and ApplyWeights private functions of the framework and call them directly at the end of RunScorePlugins.

Reply:
I like the idea of making both RunNormalizeScorePlugins and ApplyWeights private functions. However, I'd like to do it in a follow up PR where I will add unit tests for RunScorePlugins.

Reply:
SGTM